public/video-ui/src/components/VideoEdit/formComponents/VideoExpiry.js

```diff
@@ -1,27 +1,129 @@
 import React from 'react';
-import DatePicker from '../../FormFields/DatePicker';
+import Picker from 'react-datepicker';
+import moment from 'moment';
+import _ from 'lodash';
+import 'react-datepicker/dist/react-datepicker.css';
+import Icon from '../../Icon';
 
-export default class VideoExpiryEdit extends React.Component {
+function Selector({values, value, disabled, onChange}) {
+  const handler = (e) => {
+    onChange(e.target.value);
+  };
+
+  const options = values.map((value) => {
+    return <option key={value} value={value}>{value}</option>;
+  });
+
+  const params = {
+    className: "form__field form__field--select",
+    disabled: disabled,
+    value: value,
+    onChange: handler
+  };
+
+  return <select {...params}>{options}</select>;
+}
+
+function HourSelector({date, onChange}) {
+  let values = [" "];
+  if (date) {
+    values = [...new Array(24).keys()].map((hour) => moment().hour(hour).format("HH"));
+  }
 
-  onUpdateExpiry = (newDate) => {
-    const newData = Object.assign({}, this.props.video, {
-      expiryDate: newDate
-    });
+  const params = {
+    values: values,
+    value: date ? date.format("HH") : " ",
+    disabled: !date,
+    onChange: (newHour) => onChange(date.hours(newHour))
+  };
+
+  return <Selector {...params} />;
+}
+
+function MinuteSelector({date, onChange}) {
+  let values = [" "];
+  if (date) {
+    values = [0, 15, 30, 45].map((minute) => moment().minute(minute).format("mm"));
+  }
 
-    this.props.updateVideo(newData);
+  const params = {
+    values: values,
+    value: date ? date.format("mm") : " ",
```

Comment:
It makes the combo box show "empty" until you select a date from the date-picker